File: app/src/main/java/com/fishweeb/practical/TestDialogFragment.java
package com.fishweeb.practical;

import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.DialogInterface;
import android.os.Bundle;

public class TestDialogFragment extends DialogFragment
{
    public static boolean IsShown = false;

    @Override
    public Dialog onCreateDialog(Bundle saveInstanceState)
    {
        IsShown = true;
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setMessage("Confirm Pause?")
                .setPositiveButton("Yes", new DialogInterface.OnClickListener()
                {
                    @Override
                    public void onClick(DialogInterface dialog, int id)
                    {
                        IsShown = false;
                    }
                })
                .setNegativeButton("No", new DialogInterface.OnClickListener()
                {
                    @Override
                    public void onClick(DialogInterface dialog, int id)
                    {
                        IsShown = false;
                    }
                });
        return builder.create();
    }

    @Override
    public void onCancel(DialogInterface dialog)
    {
        IsShown = false;
    }

    @Override
    public void onDismiss(DialogInterface dialog)
    {
        IsShown = false;
    }

}

File: app/src/main/java/edu/androidgroupproject/TestDialogFragment.kt
package com.fishweeb.practical

import android.app.AlertDialog
import android.app.Dialog
import android.app.DialogFragment
import android.content.DialogInterface
import android.os.Bundle

class TestDialogFragment : DialogFragment() {
    override fun onCreateDialog(saveInstanceState: Bundle): Dialog {
        IsShown = true
        val builder = AlertDialog.Builder(activity)
        builder.setMessage("Confirm Pause?")
            .setPositiveButton(
                "Yes"
            ) { dialog, id ->
                IsShown =
                    false
            }
            .setNegativeButton(
                "No"
            ) { dialog, id ->
                IsShown =
                    false
            }
        return builder.create()
    }

    override fun onCancel(dialog: DialogInterface) {
        IsShown = false
    }

    override fun onDismiss(dialog: DialogInterface) {
        IsShown = false
    }

    companion object {
        var IsShown: Boolean = false
    }
}
